mesonbuild/build.py

```diff
@@ -1987,6 +1987,10 @@ def post_init(self) -> None:
             # Don't let configuration proceed with a non-static crate type
             elif self.rust_crate_type not in ['rlib', 'staticlib']:
                 raise InvalidArguments(f'Crate type "{self.rust_crate_type}" invalid for static libraries; must be "rlib" or "staticlib"')
+            # See https://github.com/rust-lang/rust/issues/110460
+            if self.rust_crate_type == 'rlib' and any(c in self.name for c in ['-', ' ']):
```

Comment:
Oh, did not notice before, but when you reverted to previous version, you also dropped that small change:

```suggestion
            if self.rust_crate_type == 'rlib' and any(c in self.name for c in {'-', ' '}):
```

Same below.

Reply:
Ah yes, went back to the wrong version.